### PR TITLE
Add an option to Kestrel to disable threadpool dispatching

### PR DESCRIPTION
**Pull request by halter73**
Up for review. I don't like awaiting KestrelThread in FrameOfT. Will create PR for feature/dev-si once this is approved.

#1390 @pakrym @cesarbs @davidfowl

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/InlineLoggingThreadPool.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        public void Cancel(TaskCompletionSource<object> tcs)
+        {
+            try
+            {
+                tcs.TrySetCanceled();
+            }
+            catch (Exception e)
+            {
+                _log.LogError(0, e, "LoggingThreadPool.Cancel");
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/InlineLoggingThreadPool.cs]: InlineLoggingThreadPool

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/InlineLoggingThreadPool.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        public void Error(TaskCompletionSource<object> tcs, Exception ex)
+        {
+            try
+            {
+                tcs.TrySetException(ex);
+            }
+            catch (Exception e)
+            {
+                _log.LogError(0, e, "LoggingThreadPool.Error");
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/InlineLoggingThreadPool.cs]: InlineLoggingThreadPool

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/InlineLoggingThreadPool.cs (hunk 1 above)
Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/InlineLoggingThreadPool.cs]: `nameof` 😄

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/InternalKestrelServerOptions.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Internal
+{
+    public class InternalKestrelServerOptions
```
Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/InternalKestrelServerOptions.cs]: As small as it is, it should still have its unit test class. Add a unit test for the default value.

Comment (halter73) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/InternalKestrelServerOptions.cs]: This class isn't intended to make it to 2.0

**Pull request merged**